packages/rocketchat-api/server/v1/rooms.js

```diff
@@ -50,6 +50,38 @@ RocketChat.API.v1.addRoute('rooms.get', { authRequired: true }, {
 	},
 });
 
+RocketChat.API.v1.addRoute('rooms.fullInfo', { authRequired: true }, {
+	get() {
+		const { updatedSince } = this.queryParams;
+		const mountRoomWithSubscription = (rooms, subscriptions) => rooms.map((room) => ({
+			room,
+			subscription: subscriptions.find((sub) => sub.rid === room._id) || {},
+		}));
+
+		let updatedSinceDate;
+		if (updatedSince) {
+			if (isNaN(Date.parse(updatedSince))) {
+				throw new Meteor.Error('error-updatedSince-param-invalid', 'The "updatedSince" query parameter must be a valid date.');
+			} else {
```

Comment:
Do you really need this `else` here?
I mean, if the `updatedSince` isn't a valid value, the `throw/error` will raise an exception and the rest of the code won't be executed, right?